### DISCUSSION TITLE
Support "application/problem+json" Content Type

### DISCUSSION THREAD

**Discussion by Maxouwell** [remix-run/react-router#11034] [Proposals]
### What version of React Router are you using?

6.17.0

### Steps to Reproduce

RFC [7807](https://datatracker.ietf.org/doc/html/rfc7807)/[9457](https://datatracker.ietf.org/doc/html/rfc9457) introduced a extensible format for json error response, and a dedicated Content-Type : "application/problem+json". 
Some server framework start to respond with this Content-Type on error (Spring Framework 6 for example).

When these responses are returned on a loader or action, the resulting response is used as text instead of a json:
([packages/router/router.ts#L3925](https://github.com/remix-run/react-router/blob/4475b56d2e18014acfb8ff5f6077ff27df98d7fe/packages/router/router.ts#L3925))

```
    let contentType = result.headers.get("Content-Type");
    // Check between word boundaries instead of startsWith() due to the last
    // paragraph of https://httpwg.org/specs/rfc9110.html#field.content-type
    if (contentType && /\bapplication\/json\b/.test(contentType)) {
      data = await result.json();
    } else {
      data = await result.text();
    }
   ```
   
Would it be possible to add the new Content Type ?

### Expected Behavior

useLoaderData or useActionData returns a Javascript object

### Actual Behavior

useLoaderData or useActionData returns a string

**Comment by timdorr** [comment]
It's probably best to wait until this has advanced beyond the Proposed Standard stage, which is the introductory level for IETF RFCs.

**Comment by brophdawg11** [comment]
I'm going to convert this to a Proposal for consideration.  However I will say that as part of our ongoing work for [single fetch](https://github.com/remix-run/remix/discussions/7640) we are already looking into on some ways to let the React Router consumer customize the response parsing: https://github.com/remix-run/react-router/pull/11029.

So I think that will eventually be your solution here - rather than RR trying to internally handle every content type under the sun.
